noxfile.py

```diff
@@ -334,4 +334,35 @@ def get_release_tag_and_commit_hash(tags: List[str]):
 
 @nox.session(python=False, name="proto-go")
 def proto_go(session: nox.Session) -> None:
+    """Generate Go bindings for protobufs."""
     session.run("./core/scripts/generate-proto.sh")
+
+
+@nox.session(name="proto-python")
+@nox.parametrize("pb", [3, 4])
+def proto_python(session: nox.Session, pb: int) -> None:
+    """Generate Python bindings for protobufs.
+
+    The pb argument is the major version of the protobuf package to use.
+
+    Tested with Python 3.10 on a Mac with an M1 chip.
+    Absolutely does not work with Python 3.7.
```

Comment:
huh, why not? or is it an arm-specific thing?

Reply:
I did run into arm-specific issues caused by the 1.47 grpcio packages, but I fixed that by using 1.48.

In Python 3.7: For pb=3, the "importlib.metadata" import fails in `wandb_internal.codegen.py`. For pb=4, `mypy-protobuf~=3.5.0` requires at least Python 3.8.

It's probably solvable by messing with the dependencies.. do we need this to work with any specific Python versions?